compiler/includer.h

```diff
@@ -0,0 +1,59 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <string>
+
+#include "flutter/fml/macros.h"
+#include "flutter/fml/mapping.h"
+#include "impeller/compiler/include_dir.h"
+#include "shaderc/shaderc.hpp"
+
+namespace impeller {
+namespace compiler {
+
+struct IncluderData {
+  std::string file_name;
+  std::unique_ptr<fml::Mapping> mapping;
+
+  IncluderData(std::string p_file_name, std::unique_ptr<fml::Mapping> p_mapping)
+      : file_name(std::move(p_file_name)), mapping(std::move(p_mapping)) {}
+};
+
+class Includer final : public shaderc::CompileOptions::IncluderInterface {
+ public:
+  Includer(std::shared_ptr<fml::UniqueFD> working_directory,
+           std::vector<IncludeDir> include_dirs,
+           std::function<void(std::string)> on_file_included);
```

Comment:
This PR made me realize we already support shader `#include`s. Nice!

Reply:
Right. I remember dragging my feet a bit on whether to wire this up then. But figured it we may end up needing it eventually and didn't want us to hack around not having this.

The includes are also added to the .depfiles generated so editing a header should also recompile any shaders that include that header.